src/utils/features_preprocessing.py

```diff
@@ -235,15 +267,15 @@ class Preprocessor:
         TF-IDF embedding combined with additional features
     """
 
-    def __init__(self,
-                 max_df: float = 0.8,
-                 min_df: float = 10,
-                 ngram_range: tuple = (1, 1),
-                 max_feat=None,
-                 norm='l2',
-                 scaler=MinMaxScaler()):
-        self.scaler=scaler
-        self._tfidf = TfidfVectorizer(max_df=max_df, min_df=min_df,ngram_range=ngram_range,max_features=max_feat,norm=norm)
+    def __init__(self, vectorizer_type=None, scaler=MinMaxScaler()):
+        if vectorizer_type == 'tfidf':
+            self.vectorizer_type = 'tfidf'
+        elif vectorizer_type == 'spacy':
+            self.vectorizer_type = 'spacy'
+        else:
+            raise ValueError("Invalid embedding type: choose 'tfidf' or 'spacy'")
+        self.scaler = scaler
+        self._tfidf = CustomTfidf()
 
```

Comment:
Thinks how to improve if statement, now it looks bad
Also, you need to change the logic for self._tfidf = CustomTfidf()
for example, vectorizer_type = ' spacy'
your if statement will pass and   self._tfidf = CustomTfidf()- will also work. But it is not spacy,